stl/inc/condition_variable

```diff
@@ -95,7 +95,9 @@ public:
 #if _HAS_CXX20
         static_assert(chrono::is_clock_v<_Clock>, "Clock type required");
 #endif // _HAS_CXX20
-        return wait_for(_Lck, _Abs_time - _Clock::now());
+        const auto _Now           = _Clock::now();
+        const _Duration _Rel_time = (_Abs_time <= _Now) ? _Duration::zero() : _Abs_time - _Now;

```

Comment:
I think it'd be better to ensure the operands have the same type.
```suggestion
        using _Common_duration = decltype(_Abs_time - _Clock::now());
        const auto _Now        = _Clock::now();
        const auto _Rel_time   = (_Abs_time <= _Now) ? _Common_duration::zero() : _Abs_time - _Now;
```

Reply:
I like `common_type_t` for this, but I can change it to `decltype` if people prefer.